mlflow/utils/virtualenv.py

```diff
@@ -32,11 +32,17 @@ def _get_mlflow_virtualenv_root():
     return os.getenv(_MLFLOW_ENV_ROOT_ENV_VAR, str(Path.home().joinpath(".mlflow", "envs")))
 
 
+_DATABRICKS_PYENV_BIN_PATH = "/databricks/.pyenv/bin/pyenv"
+
+
 def _is_pyenv_available():
     """
     Returns True if pyenv is available, otherwise False.
     """
-    return shutil.which("pyenv") is not None
+    if is_in_databricks_runtime():
+        return os.path.exists(_DATABRICKS_PYENV_BIN_PATH)
+    else:
+        return shutil.which("pyenv") is not None
```

Comment:
Just because `os.path.exists(_DATABRICKS_PYENV_BIN_PATH)` returns True doesn't necessarily mean pyenv is available, right?

Reply:
Curious why we need this special logic for Databricks.

Reply:
I think we need this because the system path used by subprocesses is not identical to the system path in the driver REPL. We've seen this in the past where `which python` in a subprocess doesn't resolve to the same python binary as the one made available to the REPL.

Reply:
Got it, let's keep this.